internal/config/flag.go

```diff
@@ -0,0 +1,67 @@
+package config
+
+import (
+	"flag"
+	"github.com/caarlos0/env/v11"
+	"log"
+)
+
+type Config struct {
+	ServerHostPort string `env:"ADDRESS"`
+	ReportInterval int    `env:"REPORT_INTERVAL"`
+	PollInterval   int    `env:"POLL_INTERVAL"`
+}
+
+var appConfig *Config
+
+func GetConfig() *Config {
+	if appConfig == nil {
+		appConfig = initAppConfig()
+	}
+	return appConfig
+}
+
+func initAppConfig() *Config {
+	var (
+		sh string
+		ri int
+		pi int
+	)
+
+	c := Config{}
+
+	err := env.Parse(&c)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
```suggestion
		return err
```
Ошибку поднимаем вверх, и там уже обрабатываем. В случае с конфигом логично падать по `log.Fatal()` при любой ошибке инициализации конфигурации.